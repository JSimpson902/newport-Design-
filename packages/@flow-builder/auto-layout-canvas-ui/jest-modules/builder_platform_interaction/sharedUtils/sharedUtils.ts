const sharedUtils = jest.requireActual('builder_platform_interaction_mocks/sharedUtils/sharedUtils');
const commands = jest.requireActual('builder_platform_interaction/sharedUtils/commands');
const lwcUtils = jest.requireActual('builder_platform_interaction/sharedUtils/lwcUtils');

const mockKeyboardInteractionUtils = jest.requireActual(
    'builder_platform_interaction_mocks/sharedUtils/keyboardInteractionUtils'
);
const auraUtils = jest.requireActual('builder_platform_interaction/sharedUtils/auraUtils');
const customIconUtils = jest.requireActual('builder_platform_interaction/sharedUtils/customIconUtils');

let keyboardInteractionUtils = jest.requireActual('builder_platform_interaction/sharedUtils/keyboardInteractionUtils');
const { KeyboardInteractions, withKeyboardInteractions } = mockKeyboardInteractionUtils;

keyboardInteractionUtils = { ...keyboardInteractionUtils, KeyboardInteractions, withKeyboardInteractions };

const { loggingUtils, storeUtils, commonUtils } = sharedUtils;

const Keys = keyboardInteractionUtils.Keys;

const focusUtils = {
    getElementWithFocus: jest.fn(),
    restoreFocus: jest.fn(),
    clearStashFocus: jest.fn(),
    stashFocus: jest.fn()
};

const appGuidanceUtils = jest.requireActual('builder_platform_interaction/sharedUtils/appGuidanceUtils');

export {
    loggingUtils,
    keyboardInteractionUtils,
    commands,
    storeUtils,
    commonUtils,
    lwcUtils,
    auraUtils,
    Keys,
    focusUtils,
    customIconUtils,
    appGuidanceUtils
};
