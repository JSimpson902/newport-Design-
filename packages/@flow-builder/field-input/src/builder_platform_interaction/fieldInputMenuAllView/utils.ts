import { getFlowElementsApi, getRulesApi } from 'builder_platform_interaction/fieldInputMenuDataApi';
import { filterAndMapToMenuItems } from 'builder_platform_interaction/fieldInputUtils';
import { ELEMENT_TYPE } from 'builder_platform_interaction/flowMetadata';
import { getLHSTypes, RULE_TYPES } from 'builder_platform_interaction/ruleLib';

/**
 * Get the sections for the "All" view
 *
 * @returns The "All" view sections
 */
export function getViewAllSections(): FieldInput.MenuSection[] {
    const config: MenuConfig = {
        sortField: 'label',
        activePicklistValues: [],
        traversalConfig: { isEnabled: true },
        filter: {
            includeNewResource: true,
            allowGlobalConstants: false,
            showSystemVariables: true,
            showGlobalVariables: true,
            shouldBeWritable: false
        }
    };

    // TODO: remove hardcoded
    const ruleType = RULE_TYPES.ASSIGNMENT;
    const elementType = ELEMENT_TYPE.DECISION;

    const rules = getRulesApi(ruleType, elementType);

    const flowELements = getFlowElementsApi();

    // TODO: find better way to wait for rules to load and remove this
    if (rules != null) {
        return filterAndMapToMenuItems(flowELements, getLHSTypes(ELEMENT_TYPE.DECISION, rules), config);
    }

    return [];
}
