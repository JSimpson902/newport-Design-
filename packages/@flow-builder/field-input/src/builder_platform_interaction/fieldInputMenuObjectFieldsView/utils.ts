/**
 *
 * @param apiData TBD
 * @returns TBD
 */
export function getObjectFieldsViewSections(apiData: GetObjectInfoApiData): FieldInput.MenuSection[] {
    const menuItems: FieldInput.MenuItem[] = Object.values(apiData.fields).map((field) => ({
        ...field,
        view:
            field.dataType === 'Picklist'
                ? {
                      type: 'PicklistValues',
                      fieldApiName: `${field.sobjectName}.${field.apiName}`,
                      recordTypeId: 'dummy'
                  } // recordTypeId can't be retrieved until we switch to UI-API
                : undefined,
        type: field.dataType,
        name: field.apiName,
        value: field.apiName,
        iconName: field.dataType === 'Picklist' ? 'utility:picklist_type' : 'utility:text',
        iconSize: 'x-small'
    }));

    return [{ items: menuItems, name: 'ObjectFields', label: undefined }];
}
