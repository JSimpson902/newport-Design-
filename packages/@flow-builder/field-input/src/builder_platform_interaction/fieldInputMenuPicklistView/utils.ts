/**
 *
 * @param apiData TBD
 * @returns TBD
 */
export function getPicklistValuesViewSection(apiData: GetPicklistValuesApiData): FieldInput.MenuSection {
    return {
        name: 'PicklistValues',
        items: (apiData.values || []).map((value) => ({
            ...value,
            label: value.label,
            name: value.value,
            key: value.value,
            iconName: 'utility:text',
            iconSize: 'x-small'
        }))
    };
}
