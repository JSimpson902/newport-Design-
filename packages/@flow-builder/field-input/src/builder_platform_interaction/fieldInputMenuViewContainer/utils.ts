const viewDispatchMap = {
    All: { isAllView: true },
    ObjectFields: { isObjectFieldsView: true },
    PicklistValues: { isPicklistValuesView: true }
} as const;

/**
 * Augments a MenuItemView with a `isXX` dispatch property
 *
 * @param view - The menu item view
 * @returns The augmented menu item view
 */
export const getViewProps = (view: FieldInput.MenuItemView) => {
    const viewDispatch = viewDispatchMap[view.type];

    return viewDispatch != null ? { ...view, ...viewDispatch } : { missingViewType: view.type };
};
