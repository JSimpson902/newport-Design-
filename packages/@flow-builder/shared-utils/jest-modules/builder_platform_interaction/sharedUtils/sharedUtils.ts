import * as loggingUtils from './loggingUtils/loggingUtils';
import * as keyboardInteractionUtils from './keyboardInteractionUtils/keyboardInteractionUtils';

import * as commands from './commands/commands';
import * as storeUtils from './storeUtils/storeUtils';
import * as commonUtils from './commonUtils/commonUtils';
export { loggingUtils, keyboardInteractionUtils, commands, storeUtils, commonUtils };
