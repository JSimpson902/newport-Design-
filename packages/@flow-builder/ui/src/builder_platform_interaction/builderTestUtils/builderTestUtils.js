export * from './screenEditorTestUtils.js';
export * from './commonTestUtils';
export * from './deepFindMatchers';
export * from './useAdvancedOptionsTestUtils';
export * from './goldObjectMatchers';
export * from './domTestUtils';
export * from './complexTypeTestUtils';
