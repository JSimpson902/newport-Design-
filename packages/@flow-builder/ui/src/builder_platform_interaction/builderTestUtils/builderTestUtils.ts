export * from './combobox';
export * from './commonTestUtils';
export * from './complexTypeTestUtils';
export * from './deepFindMatchers';
export * from './domTestUtils';
export * from './events';
export * from './goldObjectMatchers';
export * from './manuallyAssignVariablesCheckboxTestUtils';
export * from './screenEditorTestUtils';
export * from './selectors';

// @sa11y testing appears to take some extra time (beyond the given 5000ms default jest timeout)
export const ACCESSIBILITY_TEST_TIMEOUT_IN_MS = 15000;

/**
 * Mocks a label in jest
 *
 * @param jest - The jest instance
 * @param labelName - The label name
 * @returns the mock label
 */
export function mockLabel(jest, labelName: string) {
    jest.mock(
        `@salesforce/label/${labelName}`,
        () => {
            return { default: `${labelName} mock value` };
        },
        { virtual: true }
    );

    return `${labelName} mock value`;
}
