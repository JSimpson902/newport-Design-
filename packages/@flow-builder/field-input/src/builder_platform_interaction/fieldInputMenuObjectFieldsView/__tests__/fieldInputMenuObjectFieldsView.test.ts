import { createComponent } from 'builder_platform_interaction/builderTestUtils';
import { lwcUtils } from 'builder_platform_interaction/sharedUtils';
import { getObjectFieldsViewSections } from '../utils';

const tag = 'builder_platform_interaction-field-input-menu-object-fields-view';

const apiData: GetObjectInfoApiData = {
    fields: [
        {
            label: 'Address',
            apiName: 'address',
            dataType: 'string'
        },
        {
            label: 'Address2',
            apiName: 'address2',
            dataType: 'string'
        }
    ]
};

const mockApiResponse = {
    data: apiData
};

jest.mock('builder_platform_interaction/fieldInputMenuSectionItem', () =>
    jest.requireActual('builder_platform_interaction_mocks/fieldInputMenuSectionItem')
);

jest.mock('builder_platform_interaction/fieldInputMenuDataApi', () => {
    return {
        getObjectInfoApi: () => {
            return Promise.resolve(mockApiResponse);
        }
    };
});

jest.mock('../utils', () => {
    return {
        getObjectFieldsViewSections: jest.fn((apiData) =>
            jest.requireActual('../utils').getObjectFieldsViewSections(apiData)
        )
    };
});

const createComponentUnderTest = async (props) => {
    const overrideProps = {};

    return createComponent(tag, props, overrideProps);
};

const selectors = {
    ul: 'ul'
};

describe('Field Input Menu Object Fields View Tests', () => {
    let cmp;
    let dom;

    beforeEach(async () => {
        cmp = await createComponentUnderTest({ objectApiName: 'Account' });
        dom = lwcUtils.createDomProxy(cmp, selectors);
    });

    it('sanity', async () => {
        expect(cmp).toBeTruthy();
        expect(getObjectFieldsViewSections).toHaveBeenCalledTimes(1);
        expect(dom.all.ul.length).toEqual(1);
    });

    it('utils > getObjectFieldsViewSections', async () => {
        expect(getObjectFieldsViewSections(mockApiResponse.data)).toMatchSnapshot();
    });
});
