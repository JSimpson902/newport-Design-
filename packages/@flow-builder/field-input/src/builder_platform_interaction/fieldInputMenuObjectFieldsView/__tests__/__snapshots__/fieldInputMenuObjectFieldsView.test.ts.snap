// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Input Menu Object Fields View Tests utils > getObjectFieldsViewSections 1`] = `
Array [
  Object {
    "items": Array [
      Object {
        "apiName": "address",
        "dataType": "string",
        "iconName": "utility:text",
        "iconSize": "x-small",
        "label": "Address",
        "name": "address",
        "sobjectName": "Account",
        "type": "string",
        "value": "address",
        "view": undefined,
      },
      Object {
        "apiName": "address2",
        "dataType": "string",
        "iconName": "utility:text",
        "iconSize": "x-small",
        "label": "Address2",
        "name": "address2",
        "sobjectName": "Account",
        "type": "string",
        "value": "address2",
        "view": undefined,
      },
      Object {
        "apiName": "address2",
        "dataType": "Picklist",
        "iconName": "utility:picklist_type",
        "iconSize": "x-small",
        "label": "BillingGeocodeAccuracy",
        "name": "address2",
        "sobjectName": "Account",
        "type": "Picklist",
        "value": "address2",
        "view": Object {
          "fieldApiName": "Account.address2",
          "recordTypeId": "dummy",
          "type": "PicklistValues",
        },
      },
    ],
    "label": undefined,
    "name": "ObjectFields",
  },
]
`;
